generator/src/FluentStageFactoryGenerator.php

```diff
@@ -0,0 +1,163 @@
+        $namespace->addUse(ResolvesToArray::class);
+        $namespace->addUse(ResolvesToObject::class);
+        $namespace->addUse(Pipeline::class);
+        $namespace->addUse(AccumulatorInterface::class);
+        $namespace->addUse(ExpressionInterface::class);
+        $namespace->addUse(Optional::class);
+        $namespace->addUse(QueryInterface::class);
+        $namespace->addUse(Sort::class);
+        $namespace->addUse(BSONArray::class);
+        $namespace->addUse(stdClass::class);
+        $namespace->addUse(self::FACTORY_CLASS);
+        $class->addProperty('pipeline')
+            ->setType('array')
+            ->setComment('@var list<StageInterface>')
+            ->setValue([]);
+        $class->addMethod('getPipeline')
+            ->setReturnType(Pipeline::class)
+            ->setBody(<<<'PHP'
+                return new Pipeline(...$this->pipeline);
+                PHP);
+
+        $staticFactory = ClassType::from(self::FACTORY_CLASS);
+        assert($staticFactory instanceof ClassType);
+        foreach ($staticFactory->getMethods() as $method) {
+            if (! $method->isPublic()) {
+                continue;
+            }
+
+            try {
+                $this->addMethod($method, $namespace, $class);
+            } catch (Throwable $e) {
+                throw new RuntimeException(sprintf('Failed to generate class for operator "%s"', $operator->name), 0, $e);
+            }
+        }
+
+        $staticFactory = TraitType::from(Stage\FactoryTrait::class);
+        assert($staticFactory instanceof TraitType);
+        foreach ($staticFactory->getMethods() as $method) {
+            if (! $method->isPublic()) {
+                continue;
+            }
+
+            try {
+                $this->addMethod($method, $namespace, $class);
+            } catch (Throwable $e) {
+                throw new RuntimeException(sprintf('Failed to generate class for operator "%s"', $operator->name), 0, $e);
+            }
+        }
+
+        return $namespace;
+    }
+
+    private function addMethod(Method $factoryMethod, PhpNamespace $namespace, ClassType $class): void
+    {
+        if ($class->hasMethod($factoryMethod->getName())) {
+            return;
+        }
+
+        $method = $class->addMethod($factoryMethod->getName());
+
+        $method->setComment($factoryMethod->getComment());
+        $method->setParameters($factoryMethod->getParameters());
+
+        $args = array_map(
+            fn ($param) => '$' . $param->getName(),
+            $factoryMethod->getParameters(),
+        );
+
+        if ($factoryMethod->isVariadic()) {
+            $method->setVariadic();
+            $args[array_key_last($args)] = '...' . $args[array_key_last($args)];
+        }
+
+        $method->setReturnType('static');
+        $method->setBody(sprintf(
+            <<<'PHP'
+            $this->pipeline[] = %s::%s(%s);
+
+            return $this;
+            PHP,
+            'Stage',
```

Comment:
I assume this is meant to correspond to `FACTORY_CLASS`. If so, would it make sense to derive it from that instead of hard-coding its short name? That could probably be done once with:

```php
static $factoryClass = (new ReflectionClass(self::FACTORY_CLASS))->getShortName();
```